### ISSUE TITLE
tp 9, ej11

### ISSUE THREAD

**Issue by morediazmacchi** [PI-ITBA/2024_02#105]
hola, como estan? estaba tratando de hacer este ejercicio y queria consultar que tan mal esta, me tira un millon de errores de segmentation fault. Gracias

void reemplazo(int matriz[][MAX], int dim, int fil, int col){
    if(fil < 0 || fil>= dim || col < 0 || col >= dim){
        return;
    }
    else{
        matriz[fil][col] = !matriz[fil][col];
        reemplazo(matriz, dim , fil + 1 , col + 1);
        reemplazo(matriz, dim , fil-1, col-1);
        reemplazo(matriz, dim, fil+1, col-1);
        reemplazo(matriz, dim, fil-1, col+1);
    }
}

**Comment by marcelogarberoglio** [comment]
¿Te tira segFault o te dice que no pasó los asserts ?
La función reemplazo no puede ser recursiva. Una vez que negaste la coordenada original tenés que moverte por ejemplo hacia arriba a la izquierda. Pero ahí ya no tenés el problema original, porque no tenés que moverte hacia las cuatro diagonales.

**Comment by morediazmacchi** [comment]
me aparece esto: AddressSanitizer:DEADLYSIGNAL y despues 254 lineas asi #0 0x1000f78ac in reemplazo ej11.c:41, y respecto  a lo otro, entonces como seria? habia entendido como que tenia que moverme a las 4 diagonales y reemplazarlas por su negacion booleana

**Comment by marcelogarberoglio** [comment]
Sí, tiene que moverse en las 4 diagonales, pero una vez que se mueve un paso lo que tiene que hacer es seguir moviéndose en la misma dirección hasta alcanzar el borde. Esa parte puede ser recursiva.
Conclusión: la función reemplazo tiene que llamar 4 veces a una función recursiva que le diga desde dónde moverse y en qué dirección moverse. Esa función auxiliar sí puede ser recursiva
